### PR TITLE
ImageSwitcher: Add selectedIndex prop

### PR DESCRIPTION
**Pull request by tysonhummel**
For Venus https://moovweb.atlassian.net/browse/VN-355

I wasn't sure which repo to make a PR for. There is an identical PR on moov-pwa https://github.com/moovweb/moov-pwa/pull/160

### CODE REVIEW

**Comment by Seigiard** [comment]
@markbrocato We're not familiar with PRPL dev process.

What should we do to merge this PR into react-storefront? 
BTW, I can't understand why tests have been failed.

**Comment by dijs** [comment]
CI ran out of memory. We don't know the reason for that yet. I restarted the tests

File: packages/react-storefront/src/ImageSwitcher.js [hunk 1]
```diff
@@ -283,6 +288,12 @@ export default class ImageSwitcher extends Component {
     viewerActive: false
   }
 
+  componentWillReceiveProps(nextProps){
+    if (nextProps.selectedIndex !== 'undefined') {
```
Comment (markbrocato) [inline packages/react-storefront/src/ImageSwitcher.js]: @tysonhummel Do you mean `typeof nextProps.selectedIndex !== 'undefined'` here?

Comment (markbrocato) [reply packages/react-storefront/src/ImageSwitcher.js]: Also, I'd prefer you use the new `static getDerivedStateFromProps(props)` method.  componentWillReceiveProps has been deprecated.

https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

File: packages/react-storefront/src/ImageSwitcher.js [hunk 2]
```diff
@@ -283,6 +288,13 @@ export default class ImageSwitcher extends Component {
     viewerActive: false
   }
 
+  static getDerivedStateFromProps(nextProps, prevState) {
```
Comment (tysonhummel) [inline packages/react-storefront/src/ImageSwitcher.js]: @markbrocato @dijs refactored to use getDerivedStateFromProps

File: packages/react-storefront/src/ImageSwitcher.js [hunk 3]
```diff
@@ -283,6 +288,13 @@ export default class ImageSwitcher extends Component {
...
+    if (typeof nextProps.selectedIndex !== 'undefined' && nextProps.selectedIndex !== null && nextProps.selectedIndex !== prevState.selectedIndex) {
+      return { selectedIndex: nextProps.selectedIndex }
+    }
+    return {}
```
Comment (dijs) [inline packages/react-storefront/src/ImageSwitcher.js]: I think you need to return `null` if you do not want the state to change.

File: packages/react-storefront/src/ImageSwitcher.js [hunk 4]
```diff
@@ -283,6 +288,13 @@ export default class ImageSwitcher extends Component {
     viewerActive: false
   }
 
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (typeof nextProps.selectedIndex !== 'undefined' && nextProps.selectedIndex !== null && nextProps.selectedIndex !== prevState.selectedIndex) {
```
Comment (dijs) [inline packages/react-storefront/src/ImageSwitcher.js]: You could just check for a number type instead here.

File: packages/react-storefront/src/ImageSwitcher.js (hunk 3 above)
Comment (tysonhummel) [reply packages/react-storefront/src/ImageSwitcher.js]: This seems to work the same as null but I can change it.

**Comment by tysonhummel** [comment]
@markbrocato I think this is ready for another look from you.

**Comment by Seigiard** [comment]
@markbrocato Could you push this PR forward? 
Venus team wants to have it before tomorrow's launch

**Pull request merged**